### ISSUE TITLE
TC003 false-positive for dataclass.KW_ONLY when future-annotations is true

### ISSUE THREAD

**Issue by beskep** [astral-sh/ruff#21121]
### Summary

playground: https://play.ruff.rs/10ce19b3-ce84-44e2-be98-19ece9acdf9d

---

pyproject.toml
```toml
[project]
name = "TC-TEST"
version = "0.1.0"
description = "TC-TEST"
readme = "README.md"
requires-python = ">=3.13"
dependencies = ["ruff>=0.14.2"]

[tool.ruff.lint]
select = ["TC"]
extend-safe-fixes = ["TC"]
future-annotations = true
```

```python
from dataclasses import KW_ONLY, dataclass


@dataclass
class DataClass:
    a: int
    _: KW_ONLY
    b: int
```

ruff (v0.14.2) check `TC003` for importing `KW_ONLY` in python 3.13.

<br>

Auto (unsafe) fix result:
```python
from __future__ import annotations
from dataclasses import dataclass
from typing import TYPE_CHECKING
from dataclasses import KW_ONLY
```

<br>

`KW_ONLY` import in `TYPE_CHECKING` block cause `TC004`.
```python
from __future__ import annotations

from dataclasses import dataclass
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from dataclasses import KW_ONLY


@dataclass
class DataClass:
    a: int
    _: KW_ONLY
    b: int
```

Relative to #12859

### Version

ruff 0.14.2 (83a3bc4ee 2025-10-23)

**Comment by ntBre** [comment]
Interesting, thanks for the report and for finding the other issue! It seems like I probably missed something here when adding the `future-annotations` setting.
